### PR TITLE
Feat/phase out xcom backend

### PR DESCRIPTION
**Pull request by pgzmnk**
Removes the need for an XCom backend. Relevant functionality is now in `RayPythonOperator`. 

This will simplify the initial setup.

### CODE REVIEW

**Comment by dimberman** [comment]
cc: @richardliaw @grechaw

File: requirements.txt [hunk 1]
```diff
@@ -1,4 +1,5 @@
-ray[tune]==1.4.0
+ray[tune]==1.4.1
```
Comment (dimberman) [inline requirements.txt:1]: Should we bump this to 1.6.0?

File: setup.py [hunk 2]
```diff
@@ -20,7 +20,11 @@
     version=VERSION,
     packages=find_packages(include=["*"]),
     zip_safe=False,
-    install_requires=["apache-airflow>=2.0", "pandas>=1.0.0", "filelock>=3.0.0", "ray"],
+    install_requires=["apache-airflow>=2.0",
+                      "pandas>=1.0.0",
+                      "filelock>=3.0.0",
+                      "ray>=1.3.0",
```
Comment (dimberman) [inline setup.py:26]: Let's DEFINITELY bump this one. @richardliaw @grechaw what should the minimum be?

File: ray_provider/xcom/ray_backend.py [hunk 3]
```diff
@@ -31,11 +32,29 @@ def get_or_create_kv_store(identifier, allow_new=False):
 
 
 class KVStore:
+
```
Comment (dimberman) [inline ray_provider/xcom/ray_backend.py:35]: Should we even put this in an xcom folder anymore? Maybe it should exist in a utils folder?

File: README.md [hunk 4]
```diff
@@ -106,44 +105,15 @@ memory.To start, in your environment with ray installed, run:
...
 
-9. In your Airflow DAG python file, you must include the following in your
-`default_args` dictionary:
 
-    ```python
-    from ray_provider.xcom.ray_backend import RayBackend
-    .
-    .
-    .
-    default_args = {
```
Comment (dimberman) [inline README.md:117]: @pgzmnk what would happen if multiple tasks fail? Would it write the state to s3 multiple times?

File: ray_provider/xcom/ray_backend.py [hunk 5]
```diff
@@ -31,11 +32,29 @@ def get_or_create_kv_store(identifier, allow_new=False):
...
     @ray.remote
-    class _KvStoreActor(object):
-        def __init__(self):
+    class _KVStoreActor(object):
```
Comment (dimberman) [inline ray_provider/xcom/ray_backend.py:37]: @richardliaw do you foresee any problems if we launch multiple of this actor at the same time or was that more around older versions of ray?

**Review by dimberman — CHANGES_REQUESTED** [review]

File: requirements.txt (hunk 1 above)
Comment (grechaw) [reply requirements.txt:1]: Yes I think so.

File: setup.py (hunk 2 above)
Comment (grechaw) [reply setup.py:26]: Can we get away with minimum 1.5.2?

File: ray_provider/xcom/ray_backend.py (hunk 5 above)
Comment (grechaw) [reply ray_provider/xcom/ray_backend.py:37]: Is it not intended to be a singleton and hence should be guarded against having multiple copies?

Comment (dimberman) [reply ray_provider/xcom/ray_backend.py:37]: @grechaw yes but I think that the file lock should fix that right?

**Review by dimberman — APPROVED** [review]

**Pull request merged**